File: easy/392is_subsequence.cpp
// sept 6, 2022 solution

#include <iostream>
using namespace std;

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        // iterate through t skipping characters until you get s
        int j = 0;
        for (int i = 0; i < t.size(); i++)
        {
            if (t[i] == s[j])
            {
                j++;
            }
        }
        cout << j;
        if (j == (s.size()))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
};

File: easy/392is_subsequence.py
# june 27, 2023 solution

class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if s == "":
            return True
        if len(s) > len(t):
            return False
        iterator = 0
        for i in t:
            if iterator >= len(s):
                break
            elif i == s[iterator]:
                iterator += 1
        if iterator == len(s):
            return True
        return False
